src/partition/InputLines.h

```diff
@@ -68,7 +70,7 @@ class InputLines {
     /// ignoring the first line in each file. The user is responsible for
     /// ensuring that the file list contains no empty or duplicate entries.
     /// Note that `blockSize` is clamped to lie between 1MiB and 1GiB.
-    InputLines(std::vector<boost::filesystem::path> const &paths, size_t blockSize, bool skipFirstLine);
+    InputLines(std::vector<boost::filesystem::path> const &paths, size_t blockSize, bool skipFirstLine,std::vector<std::string> paramNames= {});
```

Comment:
The inefficient type of the last parameter and initialization Must be the `const` reference:
```
std::vector<std::string> const& paramNames = std::vector<std::string>()
```

Reply:
paramNames has been replaced by a struct object that contains more csv formating parameters